tests/tcp_listener.rs

```diff
@@ -72,6 +72,12 @@ fn smoke_test_tcp_listener(addr: SocketAddr) {
     thread_handle.join().expect("unable to join thread");
 }
 
+// #[test]
+// fn tcp_listener_from_std() {
```

Comment:
Maybe you can create a `net::TcpListener` (from std), set it to non-blocking convert it to a mio listener and reuse the smoke_test function (after a small rewrite)?

Reply:
Same for the other types.